Gems/GeoJSONSpawner/Code/Source/GeoJSONSpawner/GeoJSONSpawnerUtils.cpp

```diff
@@ -0,0 +1,542 @@
+                        AZ::Edit::UIHandlers::Default,
+                        &GeoJSONSpawnableAssetConfiguration::m_raytraceStartingHeight,
+                        "Raytrace starting height",
+                        "Height at which raytrace will start downwards. If Place on terrain option is set to false, this value is used as "
+                        "an altitude coordinate.");
+            }
+        }
+    }
+
+    AZStd::optional<AZ::Vector3> RaytraceTerrain(
+        const AZ::Vector3& location, const AzPhysics::SceneHandle sceneHandle, const AZ::Vector3& gravityDirection, float maxDistance)
+    {
+        AZStd::optional<AZ::Vector3> hitPosition = AZStd::nullopt;
+
+        if (sceneHandle == AzPhysics::InvalidSceneHandle)
+        {
+            return hitPosition;
+        }
+
+        auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
+        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AZ_Assert(physicsSystem, "Unable to get physics system interface.");
+
+        if (!sceneInterface || !physicsSystem)
+        {
+            return hitPosition;
+        }
+
+        AzPhysics::RayCastRequest request;
+        request.m_start = location;
+        request.m_direction = gravityDirection;
+        request.m_distance = maxDistance;
+
+        AzPhysics::SceneQueryHits result = sceneInterface->QueryScene(sceneHandle, &request);
+
+        if (!result.m_hits.empty())
+        {
+            hitPosition = result.m_hits.front().m_position;
+        }
+
+        return hitPosition;
+    }
+
+    AZ::Transform GetRandomTransform(
+        const AZ::Vector3& stdDevTranslation, const AZ::Vector3& stdDevRotation, float stdDevScale, std::mt19937& gen)
+    {
+        AZ::Transform transform = AZ::Transform::CreateIdentity();
+        AZStd::vector<std::normal_distribution<float>> distributions;
+        distributions.emplace_back(0.0f, stdDevTranslation.GetX());
+        distributions.emplace_back(0.0f, stdDevTranslation.GetY());
+        distributions.emplace_back(0.0f, stdDevTranslation.GetZ());
+        distributions.emplace_back(0.0f, stdDevRotation.GetX());
+        distributions.emplace_back(0.0f, stdDevRotation.GetY());
+        distributions.emplace_back(0.0f, stdDevRotation.GetZ());
+        distributions.emplace_back(1.0f, stdDevScale);
+
+        transform.SetTranslation(AZ::Vector3(distributions[0](gen), distributions[1](gen), distributions[2](gen)));
+        const AZ::Quaternion rotation =
+            AZ::Quaternion::CreateFromEulerAnglesDegrees(AZ::Vector3(distributions[3](gen), distributions[4](gen), distributions[5](gen)));
+        transform.SetRotation(rotation);
+        transform.SetUniformScale(AZStd::max(0.0f, distributions[6](gen)));
+        return transform;
+    }
+
+    AZStd::unordered_map<int, AZStd::vector<AzFramework::EntitySpawnTicket>> SpawnEntities(
+        AZStd::vector<GeoJSONSpawnableEntityInfo>& entitiesToSpawn,
```

Comment:
cant it be "const" as well? (and others too)

Reply:
`entitiesTosSpawn` can't be a `const` value - its data is updated in line 227:
`point = pointTransform.GetTranslation(); // Update point coordinates to use it to show labels in proper place`